lib/trip.rb

```diff
@@ -13,9 +13,25 @@ def initialize(input)
       @cost = input[:cost]
       @rating = input[:rating]
 
-      if @rating > 5 || @rating < 1
+      if @rating != nil && (@rating > 5 || @rating < 1)
         raise ArgumentError.new("Invalid rating #{@rating}")
       end
+
+      if @end_time != nil && (@start_time > @end_time)
+        raise ArgumentError.new("Invalid end time, cannot be before start time")
+      end
+    end
+
+    def trip_duration
+      if @end_time == nil
+        return 0
```

Comment:
In general, you do not need to include the name of the class in instance method names. This would be just as clear with the name `duration`.